core/src/main/scala/com/salesforce/op/filters/PreparedFeatures.scala

```diff
@@ -149,21 +158,29 @@ private[filters] object PreparedFeatures {
    * @tparam T type of the feature
    * @return tuple containing whether the feature was empty and a sequence of either doubles or strings
    */
-  private def prepareFeature[T <: FeatureType](name: String, value: T): Map[FeatureKey, ProcessedSeq] =
+  private def prepareFeature[T <: FeatureType](
+    name: String,
+    value: T,
+    timePeriod: Option[TimePeriod]): Map[FeatureKey, ProcessedSeq] =
     value match {
       case v: Text => v.value
         .map(s => Map[FeatureKey, ProcessedSeq]((name, None) -> Left(tokenize(s)))).getOrElse(Map.empty)
+      case v: Date => v.value.map { timestamp =>
```

Comment:
I wonder why Text, Date and OPNumeric are handled differently than the other values which require some value to be present `case ft@SomeValue(_)` below? @leahmcguire @marcovivero

Reply:
I believe this is because all of the values inside `case ft@SomeValue(_)` already have `ft.value` as sequences, we'll end up needing to do the same operation to unwrap the `Option[Double]`.